SYMBOL_TABLE_ENTRY_CREATION.md

```diff
@@ -0,0 +1,354 @@
+    }
+}
+```
+
+Notice the line `type etype;`. This indicates to the class that we will be setting a property type during initialization of the object. We can see that this type will be used to define the `valuesEntry`. This indicates that the values entry will be a `SymEntry` of the type `etype` and that type will be set in the `init` process. 
+
+Now look at the `proc init()`. The last parameter here is `type etype`. This requires us to pass the Chapel type into the `init` process. This type is then used to initialize the `GenSymEntry` to the proper type. Please note here that the `valuesSymEntry` must be of the type `etype` as well. 
+
+The remainder of the class is essentially the same configuration as our previous example. 
+
+<a id="createObj"></a>
+## Creating the Object
+Once you have your Symbol Table entry, you will need to configure the object that is built from the entry. This object will be used during processing on the Arkouda server. Let's look at creating this object that corresponds to our `NewSymEntry` example.
+
+First, you will need to create a new Chapel file to house the module. We will call this `NewSymObject.chpl`. Then, we will define the module in this file. Please note - we will not be highlighting the imports in this tutorial, but you will most likely need to import other Chapel/Arkouda modules.
+
+```chapel
+module NewSymObject {
+
+}
+```
+
+We will now define the class for the object. First, let's discuss some of the properties. First, we need to store the `name` associated with the `SymEntry` used to create the object. Then, we need `composite`. This is going to be our `NewSymEntry` object. We will store each entry individually as well. These objects will need to be passed to our class constructor.
+
+```chapel
+module NewSymObject {
+    class NewSymObj {
+        var name: string;
+    
+        var composite: borrowed NewSymEntry;
+    
+        var midpts: shared SymEntry(int);
+        var values: shared SymEntry(string);
+        
+        proc init(entryName:string, entry:borrowed NewSymEntry) {
+            name = entryName;
+            composite = entry;
+            
+            midpts = composite.midptEntry;
+            values = composite.valuesEntry;
+        }
+    }
+}
+```
+
+Now that we have our object, we need to add methods to create it from Chapel arrays. These methods will create the required `SymEntry` objects and use them to generate a `NewSymEntry` object. Let's update our module to add this process.
+
+```chapel
+module NewSymObject {
+    proc getNewSymObj(midpts: [] int, values: [] string, st: borrowed SymTab): owned NewSymObj throws {
+        var midptEntry = new shared SymEntry(midpts);
+        var valuesEntry = new shared SymEntry(values);
+        var newEntry = new shared NewSymEntry(midptEntry, valuesEntry);
+        var name = st.nextName();
+        st.addEntry(name, segEntry);
+        return getNewSymObj(name, st, segEntry.etype);
```

Comment:
I think instead of `segEntry` in these 2 spots we want `newEntry`

Reply:
updated